packages/grid/x-data-grid-pro/src/tests/columnHeaders.DataGridPro.test.tsx

```diff
@@ -235,14 +236,14 @@ describe('<DataGridPro /> - Column Headers', () => {
       clock.runToLast();
 
       const menu = screen.queryByRole('menu');
-      const unsortMenuitem = screen.queryByRole('menuitem', { name: /unsort/i });
+      const descMenuitem = screen.queryByRole('menuitem', { name: /sort by desc/i });
       expect(menu).toHaveFocus();
 
       fireEvent.keyDown(menu, { key: 'ArrowDown' });
-      expect(unsortMenuitem).toHaveFocus();
-      fireEvent.keyDown(unsortMenuitem, { key: 'Enter' });
-
-      expect(columnCell).toHaveFocus();
+      expect(descMenuitem).toHaveFocus();
+      // TODO: Fix
```

Comment:
```diff
-    // Emit menuOpen or menuClose events
-    const eventName = open ? 'menuOpen' : 'menuClose';
-    apiRef.current.publishEvent(eventName, { target });
```
Should we stop emitting these events? Is there a possibility somebody is using them in their logic?

Reply:
Thanks a lot @cherniavskii for jumping in for the solution 🙏, it works perfectly. I was not sure if we should directly access apiRef.current.state. Regarding events, I kept emitting them just in case they are useful for somebody. If there's something I am not getting as the reason to remove them, kindly let me know.

There was another failing use-case when we had a grouping column and clicked `Stop grouping by x`, I managed to get it working by falling back to the first visible header row.
```diff
diff --git a/packages/grid/x-data-grid/src/hooks/features/columnMenu/useGridColumnMenu.ts b/packages/grid/x-data-grid/src/hooks/features/columnMenu/useGridColumnMenu.ts
index 931d2bb41..2b5469965 100644
--- a/packages/grid/x-data-grid/src/hooks/features/columnMenu/useGridColumnMenu.ts
+++ b/packages/grid/x-data-grid/src/hooks/features/columnMenu/useGridColumnMenu.ts
@@ -47,4 +46,14 @@ export const useGridColumnMenu = (
   );
 
   const hideColumnMenu = React.useCallback<GridColumnMenuApi['hideColumnMenu']>(() => {
     const field = apiRef.current.state.columnMenu.field;
+    const columnLookup = apiRef.current.state.columns.lookup;
+    // next visible column to fallback to if current column gets hidden
+    const fallbackField = apiRef.current.getVisibleColumns()[0]?.field || field;
     if (field) {
+      apiRef.current.setColumnHeaderFocus(columnLookup[field] ? field : fallbackField);
     }

     const shouldUpdate = apiRef.current.setState((state) => {
       if (!state.columnMenu.open && state.columnMenu.field === undefined) {
```
Let me know if a better solution comes to your mind.

Reply:
> I was not sure if we should directly access apiRef.current.state

Good point, we should use selectors where possible:
```diff
- const field = apiRef.current.state.columnMenu.field;
+ const columnMenuState = gridColumnMenuSelector(apiRef.current.state);
```

Reply:
> Regarding events, I kept emitting them just in case they are useful for somebody. If there's something I am not getting as the reason to remove them, kindly let me know.

I don't see a reason to remove the column menu open/close events. Were there any changes in this PR that make them less useful?

Reply:
> Were there any changes in this PR that make them less useful?

I don't think so, I feel they might be useful in some use-cases

Reply:
I just realized that I removed those events in the diff I suggested above 🙃
No need to remove them, thanks for adding them back!

Reply:
Ah ok 😌 I thought you have some reason in mind for removing them which I couldn't guess 😄